src/HotChocolate/Data/src/Data/Projections/Expressions/QueryableProjectionVisitor.cs

```diff
@@ -24,13 +26,21 @@ public class QueryableProjectionVisitor : ProjectionVisitor<QueryableProjectionC
                 return Continue;
             }
 
-            context.PushInstance(Expression.Convert(context.GetInstance(), objectType.RuntimeType));
-            scope.Level.Push(new Queue<MemberAssignment>());
+            context.PushInstance(
+                Expression.Convert(context.GetInstance(), objectType.RuntimeType));
+            scope.Level.Push(new());
 
             var res = base.VisitObjectType(field, objectType, selection, context);
 
             context.PopInstance();
-            scope.AddAbstractType(objectType.RuntimeType, scope.Level.Pop());
+
+            {
+                var initializers = scope.Level.Pop();
+                var initializersWithType = ProjectedValue.AppendObjectType(
+                    initializers, objectType);
+
+                scope.AddAbstractType(objectType.RuntimeType, initializersWithType);
+            }
```

Comment:
this i mean 

```suggestion

                var initializers = scope.Level.Pop();
                var initializersWithType = ProjectedValue.AppendObjectType(
                    initializers, objectType);

                scope.AddAbstractType(objectType.RuntimeType, initializersWithType);
```

Reply:
I just do this to logically group lines of code, and also provide a clearer scope for the locals. Are you opposed to it?

Reply:
i guess a comment would do too, at the moment it just looks like someone forgot to remove the brackes after a condition was removed

Reply:
I tend to do that quite often in my code, and consider this a good practice. I can remove it if you don't like it.